src/meta/user/authorizer.ts

```diff
@@ -113,8 +112,46 @@ const canEdit = (props: {
   return false
 }
 
+/**
+ * CanEditAssessmentStatus
+ * Viewer or non loggedin user: never
+ * Administrator: always
+ * NationalCorrespondant and AlternateNationalCorrespondant:
+ * if status is editing then true
+ * Reviewer:
+ * if status in status ('review','editing') then true
+ * @param props
+ * @param props.country
+ * @param props.cycle
+ * @param props.user
+ * @returns boolean
+ */
+const canEditAssessmentStatus = (props: { country: Country; cycle: Cycle; user: User }): boolean => {
+  const { country, cycle, user } = props
+  const { countryIso } = country
+  const { status } = country.props
+
+  if (!user) return false
+  if (Users.isViewer(user, countryIso, cycle)) return false
+  if (Users.isAdministrator(user)) return true
+
+  if (
+    (Users.isNationalCorrespondent(user, countryIso, cycle) ||
+      Users.isAlternateNationalCorrespondent(user, countryIso, cycle)) &&
+    status === AssessmentStatus.editing
+  )
+    return true
+
+  if (Users.isReviewer(user, countryIso, cycle)) {
+    return [AssessmentStatus.editing, AssessmentStatus.review].includes(status)
+  }
+
```

Comment:
Don't we have A centralized place
To check status allowance?

Reply:
nope, we don't :(

Reply:
Can't we get status allowed to edit and then check if the status passed to the middleware is included in those ?

Reply:
mmmm, can we do that in another pr? at the moment the status check is used by canEditData and canEditCountryProps and the logic is "check the role then check the status"
we have to reverse that logic "get the roles allowed by the status then check the role"

Reply:
sure, let's talk in the next daily